### PR TITLE
ENG-1209: Add cargo check and eslint linting 

### PR DESCRIPTION
**Pull request by timmyjose**
Changes:
  - added `Cargo` check for all jobs.

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -38,6 +41,26 @@ jobs:
         echo "Waiting for instance to be ready..."
         aws ec2 wait instance-running --instance-ids ${INSTANCE_ID}
 
+  eslint:
```
Comment (mateuszmlc) [inline .github/workflows/benchmarks.yml:44]: I think it's better if we move the eslint job to a separate workflow, then we won't need to run this heavy workflow on changes to `site`

**Review by mateuszmlc — CHANGES_REQUESTED** [review]

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (timmyjose) [reply .github/workflows/benchmarks.yml:44]: Makes sense. I'll work on that 👍 .

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -55,6 +55,10 @@ jobs:
         override: true
 
     - uses: Swatinem/rust-cache@v2
+
+    - name: Cargo check
```
Comment (mateuszmlc) [inline .github/workflows/benchmarks.yml:59]: We can also probably remove the cargo checks from the benchmarks workflow, just having it in `test` should be enough

Comment (timmyjose) [reply .github/workflows/benchmarks.yml]: Done!

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -43,6 +47,10 @@ jobs:
 
     - uses: Swatinem/rust-cache@v2
 
+    - name: Cargo fmt check
+      run:  cargo fmt --check
+      working-directory: ./polylang
```
Comment (mateuszmlc) [inline .github/workflows/test.yml:52]:
```suggestion
      working-directory: ./miden
```

Copy paste error?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -74,6 +82,10 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Cargo fmt check
+        run:  cargo fmt --check
+        working-directory: ./polylang
```
Comment (mateuszmlc) [inline .github/workflows/test.yml:87]:
```suggestion
        working-directory: ./risc-zero
```

Same here

**Review by mateuszmlc — APPROVED** [review]
Looks fine otherwise

File: .github/workflows/site_lints.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+
+      - name: Install dependencies
+        run: npm install
```
Comment (calummoore) [inline .github/workflows/site_lints.yml:23]: Might be better to use yarn, as that's what the lock uses

**Review by calummoore — CHANGES_REQUESTED** [review]

File: .github/workflows/site_lints.yml (hunk 5 above)
Comment (timmyjose) [reply .github/workflows/site_lints.yml]: Done!

**Review by calummoore — APPROVED** [review]

**Pull request merged**